lib/mempool/mempool.c

```diff
@@ -73,14 +73,14 @@ static size_t buf_size(struct sys_mem_pool_base *p)
 
 static bool block_fits(struct sys_mem_pool_base *p, void *block, size_t bsz)
 {
-	return (block + bsz - 1 - p->buf) < buf_size(p);
+	return ((u8_t *)block + bsz - 1 - (u8_t *)p->buf) < buf_size(p);
 }
 
 void _sys_mem_pool_base_init(struct sys_mem_pool_base *p)
 {
 	int i;
 	size_t buflen = p->n_max * p->max_sz, sz = p->max_sz;
-	u32_t *bits = p->buf + buflen;
+	u32_t *bits = (u32_t *)((u8_t *)p->buf + buflen);
```

Comment:
can't you use `uintptr_t` here ? It's more portable that `u32_t`

Reply:
I don't see why. It has to become a pointer-to-u32_t regardless.
In this translation unit, u32_t is fine from a portability perspective as Zephyr promises to make that type available.